clang/test/CIR/CodeGen/fp-math-precision-opts.c

```diff
@@ -0,0 +1,99 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-cir -fmath-errno %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s -check-prefix=CIR-ERRNO
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-cir %s -o %t-no-errno.cir
+// RUN: FileCheck --input-file=%t-no-errno.cir %s -check-prefix=CIR-NO-ERRNO
+
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-cir -fmath-errno -O1 %s -o %t-errno-o1.cir
+// RUN: FileCheck --input-file=%t-errno-o1.cir %s -check-prefix=CIR-ERRNO-O1
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-cir -ffast-math -O1 %s -o %t-no-errno-o1.cir
+// RUN: FileCheck --input-file=%t-no-errno-o1.cir %s -check-prefix=CIR-NO-ERRNO-O1
+
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-llvm -fmath-errno %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=LLVM-ERRNO
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-llvm -ffast-math %s -o %t-no-errno.ll
+// RUN: FileCheck --input-file=%t-no-errno.ll %s -check-prefix=LLVM-NO-ERRNO
+
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-llvm -fmath-errno -O1 %s -o %t-errno-o1.ll
+// RUN: FileCheck --input-file=%t-errno-o1.ll %s -check-prefix=LLVM-ERRNO-O1
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-llvm -ffast-math -O1 %s -o %t-no-errno-o1.ll
+// RUN: FileCheck --input-file=%t-no-errno-o1.ll %s -check-prefix=LLVM-NO-ERRNO-O1
+
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -emit-llvm -fmath-errno %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=OGCG-ERRNO
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -emit-llvm -ffast-math %s -o %t-no-errno.ll
+// RUN: FileCheck --input-file=%t-no-errno.ll %s -check-prefix=OGCG-NO-ERRNO
+
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -emit-llvm -fmath-errno -O1 %s -o %t-errno-o1.ll
+// RUN: FileCheck --input-file=%t-errno-o1.ll %s -check-prefix=OGCG-ERRNO-O1
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -emit-llvm -ffast-math -O1 %s -o %t-no-errno-o1.ll
+// RUN: FileCheck --input-file=%t-no-errno-o1.ll %s -check-prefix=OGCG-NO-ERRNO-O1
+
+float test_normal(float f) {
+  return __builtin_cosf(f);
+  // CIR-ERRNO: test_normal
+  // CIR-ERRNO: cir.call @cosf
+  // CIR-NO-ERRNO: test_normal
+  // CIR-NO-ERRNO: cir.cos
+  // LLVM-ERRNO: test_normal
+  // LLVM-ERRNO: call float @cosf
+  // LLVM-NO-ERRNO: test_normal
+  // LLVM-NO-ERRNO: call float @llvm.cos.f32
+  // OGCG-ERRNO: test_normal
+  // OGCG-ERRNO: call {{.*}} float @cosf
+  // OGCG-NO-ERRNO: test_normal
+  // OGCG-NO-ERRNO: call {{.*}} float @llvm.cos.f32
```

Comment:
```suggestion
  // ALL: test_normal
  // CIR-ERRNO: cir.call @cosf
  // CIR-NO-ERRNO: cir.cos
  // LLVM-ERRNO: call float @cosf
  // LLVM-NO-ERRNO: call float @llvm.cos.f32
  // OGCG-ERRNO: call {{.*}} float @cosf
  // OGCG-NO-ERRNO: call {{.*}} float @llvm.cos.f32
```
You can do this and then modify the RUN lines to replace `--check-prefix=CIR-ERRNO` (for example) with `--check-prefixes=ALL,CIR-ERRNO`